querybook/server/lib/query_analysis/validation/validators/presto_optimizing_validator.py

```diff
@@ -171,53 +214,117 @@ def get_query_validation_results(
                 like_strings = []
             token_idx += 1
 
-        return validation_errors
+        return validation_results
 
 
-class PrestoOptimizingValidator(BaseQueryValidator):
-    def languages(self):
-        return ["presto", "trino"]
+class ColumnNameSuggester(BasePrestoSQLGlotDecorator):
+    @property
+    def message(self):
+        return ""  # Unused, message is not changed
 
-    def _get_explain_validator(self):
-        return PrestoExplainValidator("")
+    @property
+    def severity(self):
+        return QueryValidationSeverity.WARNING  # Unused, severity is not changed
 
-    def _get_sqlglot_validators(self) -> List[BaseSQLGlotValidator]:
-        return [
-            UnionAllValidator(),
-            ApproxDistinctValidator(),
-            RegexpLikeValidator(),
-        ]
+    def _get_tables_in_query(self, query: str, engine_id: int) -> List[str]:
+        engine = get_query_engine_by_id(engine_id)
+        tables_per_statement, _ = process_query(query, language=engine.language)
+        return list(chain.from_iterable(tables_per_statement))
 
-    def _get_sql_glot_validation_results(
-        self, query: str
-    ) -> List[QueryValidationResult]:
-        validation_suggestions = []
-
-        query_raw_tokens = None
-        for validator in self._get_sqlglot_validators():
-            if query_raw_tokens is None:
-                query_raw_tokens = validator._tokenize_query(query)
-            validation_suggestions.extend(
-                validator.get_query_validation_results(
-                    query, raw_tokens=query_raw_tokens
-                )
-            )
+    def _is_column_name_error(self, validation_result: QueryValidationResult) -> bool:
+        return bool(
+            re.search(r"Column .* cannot be resolved", validation_result.message)
+        )
 
-        return validation_suggestions
+    def _get_column_name_from_position(
+        self, query: str, start_line: int, start_ch: int, raw_tokens: List[Token]
+    ) -> str:
+        if raw_tokens is None:
+            raw_tokens = self._tokenize_query(query)
+        column_error_start_index = self._get_query_index_by_coordinate(
+            query, start_line, start_ch
+        )
+        for token in raw_tokens:
+            if token.start == column_error_start_index:
+                return token.text
+        return None
 
-    def _get_presto_explain_validation_results(
-        self, query: str, uid: int, engine_id: int
-    ) -> List[QueryValidationResult]:
-        return self._get_explain_validator().validate(query, uid, engine_id)
+    def _search_columns_for_suggestion(self, columns: List[str], suggestion: str):
+        """Return the case-sensitive column name by searching the table's columns for the suggestion text"""
```

Comment:
isn't the highlighted column name always one of the columns? wondering if this function is necessary.

Reply:
The highlighted column is always lowercase, this makes sure we get the case-sensitive version of the column name